File: Data Structure and Algorithms I/Carrillo Cervantes Ivette Alejandra G1 P12 V2/actividad2.c
// Carrillo Cervantes Ivette Alejandra
// Juárez Juárez María José
// Estructura de Datos y Algoritmos I

#include <stdio.h>

int cat(int num){
	if (num%2==0)
	{
		num--;
	}
	if (num>0)
	{
		printf("%d\n", num);
		cat(num-2);
	}
}

void main(){
	int num;
	printf("\nIngrese un numero: ");
	scanf("%d", &num);
	cat(num);
}

File: Data Structure and Algorithms I/Carrillo Cervantes Ivette Alejandra G1 P11 V2/actividad2.py
"""
CARRILLO CERVANTES IVETTE ALEJANDRA
ESTRUCTURA DE DATOS Y ALGORITMOS - GRUPO 1
"""

def activities(s, f, n): # FUNCION ACTIVIDADES, TIENE COMO PARAMETROS LA LISTA s, f Y LA VARIABLE n
	print('\nSelected activities are:\n')
	i = 0
	print(f'A{i+1}')
	j = 1
	inicio = 1
	incremento = 1
	for j in range(inicio, n, incremento): # CICLO FOR
		if s[j]>=f[i]: # SI EL SEGUNDO INDICE DE s ES MAYOR O IGUAL AL PRIMER INDICE DE f
			print(f'A{j+1}') # IMPRIME EL HORARIO
			i = j # i SE IGUALA A j

s = [1, 2, 3, 2, 4, 5, 6, 8, 7] # horarios inicio de actividades
f = [4, 5, 6, 8, 6, 7, 7, 12, 9] # horario fin de actividades
n = len(s)
activities(s, f, n)